src/main/java/io/tus/java/client/TusClient.java

```diff
@@ -32,14 +35,36 @@ public TusClient() {
     }
 
     /**
-     * Set the URL used for creating new uploads. This is required if you want to initiate new
+     * Set the URL used for creating new uploads. This or setUploadLocationURL is required if you want to initiate new
      * uploads using {@link #createUpload} or {@link #resumeOrCreateUpload} but is not used if you
-     * only resume existing uploads.
+     * only resume existing uploads. Will enable the use of the Creation Extension when creating uploader
      *
      * @param uploadCreationURL Absolute upload creation URL
      */
     public void setUploadCreationURL(URL uploadCreationURL) {
         this.uploadCreationURL = uploadCreationURL;
+        this.useCreationExtension = true;
+    }
+
+    /**
+     * Set the URL used for creating new uploads. This or setUploadLocationURL is required if you want to initiate new
```

Comment:
I believe this comment belongs to setUploadCreationURL instead. Please correct it